openhands/tools/planning_file_editor/definition.py

```diff
@@ -0,0 +1,116 @@
+"""Planning file editor tool - combines read-only viewing with PLAN.md editing."""
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from openhands.sdk.conversation.state import ConversationState
+
+from openhands.sdk.tool import (
+    ToolAnnotations,
+    ToolDefinition,
+)
+from openhands.tools.file_editor.definition import (
+    TOOL_DESCRIPTION as FILE_EDITOR_TOOL_DESCRIPTION,
+    FileEditorAction,
+    FileEditorObservation,
+)
+
+
+# Hardcoded plan filename
+PLAN_FILENAME = "PLAN.md"
+
+
+class PlanningFileEditorAction(FileEditorAction):
+    """Schema for planning file editor operations.
+
+    Inherits from FileEditorAction but restricts editing to PLAN.md only.
+    Allows viewing any file but only editing PLAN.md.
+    """
+
+    kind: str = "PlanningFileEditorAction"
+
+
+class PlanningFileEditorObservation(FileEditorObservation):
+    """Observation from planning file editor operations.
+
+    Inherits from FileEditorObservation - same structure, just different type.
+    """
+
+    kind: str = "PlanningFileEditorObservation"
```

Comment:
Why can't we just re-use `FileEditorAction`/`FileEditorObservation` here?

Reply:
In that case, i think we shouldn't override kind which will be setted by the discriminative union mixin

Reply:
In that case, i think we shouldn't override kind which will be setted by the discriminative union mixin